### PR TITLE
feat(share/availability): extend SharesAvailable method with peer param

### PR DESCRIPTION
**Pull request by vgonkivs**
## Overview
Resolves https://github.com/celestiaorg/celestia-node/issues/1450

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: nodebuilder/share/share.go [hunk 1]
```diff
@@ -43,8 +45,8 @@ type Module interface {
 // TODO(@distractedm1nd): These structs need to be autogenerated.
 type API struct {
 	Internal struct {
-		SharesAvailable           func(context.Context, *share.Root) error `perm:"public"`
-		ProbabilityOfAvailability func(context.Context) float64            `perm:"public"`
+		SharesAvailable           func(context.Context, *share.Root, ...peer.ID) error `perm:"public"`
```
Comment (distractedm1nd) [inline nodebuilder/share/share.go:48]: I'm not sure variadic params work with the RPC, but I will check for you. Worst case is it has to take an array

**Review by distractedm1nd — COMMENTED** [review]
the godocs still have to be updated to explain the usage of peers as a param

File: nodebuilder/share/share.go [hunk 2]
```diff
@@ -26,9 +29,8 @@ var _ Module = (*API)(nil)
 //
 //go:generate mockgen -destination=mocks/api.go -package=mocks . Module
 type Module interface {
-	// SharesAvailable subjectively validates if Shares committed to the given Root are available on
-	// the Network.
-	SharesAvailable(context.Context, *share.Root) error
+	// SharesAvailable subjectively validates if Shares committed to the given Root are available on the Network.
+	SharesAvailable(context.Context, *share.Root, ...peer.ID) error
```
Comment (Wondertan) [inline nodebuilder/share/share.go:33]: Hmm, it's not necessary to bring peering to the module level, so let's avoid that and simplify things

File: nodebuilder/share/share.go [hunk 3]
```diff
@@ -87,3 +86,30 @@ func (api *API) GetSharesByNamespace(
 ) ([]share.Share, error) {
 	return api.Internal.GetSharesByNamespace(ctx, root, namespace)
 }
+
+type module struct {
+	shareService *service.ShareService
+}
```
Comment (Wondertan) [inline nodebuilder/share/share.go:92]: optional nit: we can embed it and only override `SharesAvailable`, but that's very optional

**Review by Wondertan — APPROVED** [review]

File: nodebuilder/share/share.go (hunk 3 above)
Comment (vgonkivs) [reply nodebuilder/share/share.go]: already pushed when saw your comment 😄

File: nodebuilder/share/mocks/api.go [hunk 4]
```diff
@@ -83,17 +82,17 @@ func (mr *MockModuleMockRecorder) GetSharesByNamespace(arg0, arg1, arg2 interfac
 }
 
 // ProbabilityOfAvailability mocks base method.
-func (m *MockModule) ProbabilityOfAvailability() float64 {
+func (m *MockModule) ProbabilityOfAvailability(arg0 context.Context) float64 {
```
Comment (Wondertan) [inline nodebuilder/share/mocks/api.go:85]: Was this missed by past PR?

Comment (Wondertan) [reply nodebuilder/share/mocks/api.go:85]: Seems like it's unrelated

**Review by distractedm1nd — APPROVED** [review]
great, also now no worries with the RPC 👍🏼

**Review by Wondertan — APPROVED** [review]

**Review by walldiss — APPROVED** [review]

**Review by renaynay — APPROVED** [review]

**Pull request merged**